### PR TITLE
post questions

### PR DESCRIPTION
**Pull request by mahmoudhammad309**
create the studentAuth middleware and create the route to post questions

### CODE REVIEW

**Review by Bas-Shiekh — APPROVED** [review]

File: server/controllers/class/postQuestion.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Request, Response, NextFunction } from 'express';
+import { postQuestionQuery } from '../../queries/class';
+
+const postQuestion = async (req:Request, res:Response, next:NextFunction) => {
+  try {
+    const { classId } = req.params;
+    const { question } = req.body;
+
+    await postQuestionQuery(classId, question);
+    res.json({ msg: 'added successfully' });
```
Comment (moustf) [inline server/controllers/class/postQuestion.ts:10]: Would you consider returning the data to the front end in the response?

File: server/queries/class/postQuestionQuery.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Question } from '../../models';
+
+const postQuestionQuery = async (classId: string, question: string) => {
+  await Question.create({
```
Comment (moustf) [inline server/queries/class/postQuestionQuery.ts:4]: Here you are just executing the query, try returning the query and await it in the controller.

**Review by moustf — DISMISSED** [review]

File: server/controllers/class/postQuestion.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Request, Response, NextFunction } from 'express';
+import { postQuestionQuery } from '../../queries/class';
+
+const postQuestion = async (req:Request, res:Response, next:NextFunction) => {
+  try {
+    const { classId } = req.params;
+    const { question } = req.body;
```
Comment (Raghad-Mezied) [inline server/controllers/class/postQuestion.ts:7]: Validate the body before using it

File: server/controllers/class/postQuestion.ts (hunk 1 above)
Comment (Raghad-Mezied) [inline server/controllers/class/postQuestion.ts:10]: When adding a new resource to your database, then the status code is 201.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/queries/class/postQuestionQuery.ts (hunk 2 above)
Comment (Raghad-Mezied) [reply server/queries/class/postQuestionQuery.ts:4]: Good Job Mustafa!

File: server/controllers/class/postQuestion.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    const data = {
+      id: isData.getDataValue('id'),
+      question: isData.getDataValue('question'),
+      createdAt: isData.getDataValue('createdAt'),
+    };
+
+    res.status(201).json({ msg: 'added question successfully', data });
+  } catch (error) {
+    if (error.name === 'ValidationError') next(new CustomError(400, 'question is required'));
```
Comment (Raghad-Mezied) [inline server/controllers/class/postQuestion.ts:21]: If not? handle the other errors(pass them to the error middleware)

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by Aaya-Elsharief — DISMISSED** [review]

File: server/routes/class.ts [hunk 5]
```diff
@@ -34,5 +35,6 @@ classRouter.delete('/class/assignment/:id', userAuth, teacherAuth, deleteAssignm
 classRouter.post('/class/:classId/announcement', userAuth, teacherAuth, addAnnouncement);
 
 classRouter.put('/class/:classId/questions/:questionId', userAuth, teacherAuth, putAnswerQuestion);
+classRouter.post('/class/:classId/questions', userAuth, studentAuth, postQuestion);
```
Comment (moustf) [inline server/routes/class.ts:39]: Change the endpoints to follow the convention we agreed at!

**Review by moustf — CHANGES_REQUESTED** [review]

**Review by Bas-Shiekh — APPROVED** [review]

**Review by moustf — APPROVED** [review]

**Review by Aaya-Elsharief — APPROVED** [review]

**Review by Bas-Shiekh — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Pull request merged**